### ISSUE TITLE
Improve UI/UX for users being blocked mid-conversation.

### ISSUE THREAD

**Issue by alsutton** [mastodon/mastodon#23208]
### Pitch

At the moment if there is a conversation between A and B in posted replies, either party can block the other bringing the conversation to a close. This behaviour is understandable, and, in my view, the right thing to happen.

The problem I can see is that, from an external users perspective (User C), it is not obvious why the conversation stopped. This can create the impression that one user stopped engaging in the conversation, when, in reality, they were unable to continue the conversation due to being blocked, and so they were effectively silenced.

If A blocked B then it would be useful to see a footer on posts between them saying "B can not reply to this post due to being blocked by A", which makes it clear why the conversation stopped.

### Motivation

"Reply and Block" is a toxic behaviour when the "reply" appears to continue the conversation but the author blocks the other party from responding by blocking them. 

Clearly indicating the cause of the end of the conversation will allow external viewers to get more context on why the conversation ended.

**Comment by smiba** [comment]
Potential duplicate of #23207 -- Would make more sense to combine these

**Comment by alsutton** [comment]
To me I can see two phases to the development; The first gives a better indication of why the user can't reply (which is #23207), and this one which shows others why a user is unable to reply to a post which mentions them.
